File: solutions/zig/04-pn5/code/src/main.zig
const std = @import("std");

var stdin_buffer: [4096]u8 = undefined;
var stdin_reader = std.fs.File.stdin().readerStreaming(&stdin_buffer);
const stdin = &stdin_reader.interface;

var stdout_writer = std.fs.File.stdout().writerStreaming(&.{});
const stdout = &stdout_writer.interface;

pub fn main() !void {
    while (true) {
        try stdout.print("$ ", .{});

        const command = try stdin.takeDelimiter('\n');

        if (std.mem.eql(u8, command.?, "exit")) {
            break;
        }

        try stdout.print("{s}: command not found\n", .{command.?});
    }
}

File: solutions/haskell/04-pn5/code/app/Main.hs
module Main (main) where

import System.IO (hFlush, stdout)

main :: IO ()
main = do
    putStr "$ "
    hFlush stdout
    command <- getLine
    if command == "exit"
        then pure ()
        else do
            putStrLn $ command ++ ": command not found"
            main
